Upstream-Status: Backport

Fixes the following build error:
../../chrome/browser/extensions/api/tabs/tabs_api.cc: In member function 'virtual bool extensions::WindowsCreateFunction::RunSync()':
../../chrome/browser/extensions/api/tabs/tabs_api.cc:523:8: error: variable 'saw_focus_key' set but not used [-Werror=unused-but-set-variable]
   bool saw_focus_key = false;
        ^
../../chrome/browser/extensions/api/tabs/tabs_api.cc: At global scope:
../../chrome/browser/extensions/api/tabs/tabs_api.cc:263:6: error: 'bool extensions::{anonymous}::IsHangoutsExtensionId(const string&)' defined but not used [-Werror=unused-function]
 bool IsHangoutsExtensionId(const std::string& extension_id) {
      ^
cc1plus: all warnings being treated as errors

Signed-off-by: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
---

From e428744d02cbd5a6b479705b0ba7e92dd4d934f0 Mon Sep 17 00:00:00 2001
From: "a.obzhirov" <a.obzhirov@samsung.com>
Date: Fri, 26 Aug 2016 13:19:46 -0700
Subject: [PATCH] Build error in
 chrome/browser/extensions/api/tabs/tabs_api.cc.

BUG=

Review-Url: https://codereview.chromium.org/2275113003
Cr-Commit-Position: refs/heads/master@{#414793}
---
 chrome/browser/extensions/api/tabs/tabs_api.cc | 20 ++++++++------------
 1 file changed, 8 insertions(+), 12 deletions(-)

diff --git a/chrome/browser/extensions/api/tabs/tabs_api.cc b/chrome/browser/extensions/api/tabs/tabs_api.cc
index 093bd9f..f4d7aa5 100644
--- a/chrome/browser/extensions/api/tabs/tabs_api.cc
+++ b/chrome/browser/extensions/api/tabs/tabs_api.cc
@@ -260,14 +260,6 @@ bool IsValidStateForWindowsCreateFunction(
   return true;
 }
 
-bool IsHangoutsExtensionId(const std::string& extension_id) {
-  for (const char* id : extension_misc::kHangoutsExtensionIds) {
-    if (extension_id == id)
-      return true;
-  }
-  return false;
-}
-
 }  // namespace
 
 void ZoomModeToZoomSettings(ZoomController::ZoomMode zoom_mode,
@@ -516,11 +508,11 @@ bool WindowsCreateFunction::RunSync() {
 
 #if defined(USE_ASH)
   bool create_ash_panel = false;
+  bool saw_focus_key = false;
 #endif  // defined(USE_ASH)
 
   gfx::Rect window_bounds;
   bool focused = true;
-  bool saw_focus_key = false;
   std::string extension_id;
 
   if (create_data) {
@@ -539,9 +531,11 @@ bool WindowsCreateFunction::RunSync() {
       // Only ChromeOS' version of chrome.windows.create would create a panel
       // window. It is whitelisted to Hangouts extension for limited time until
       // it transitioned to other types of windows.
-        if (IsHangoutsExtensionId(extension_id)) {
-          create_ash_panel = true;
-          break;
+        for (const char* id : extension_misc::kHangoutsExtensionIds) {
+          if (extension_id == id) {
+            create_ash_panel = true;
+            break;
+          }
         }
 #endif  // defined(USE_ASH)
         // Everything else gets POPUP instead of PANEL.
@@ -593,7 +587,9 @@ bool WindowsCreateFunction::RunSync() {
 
     if (create_data->focused) {
       focused = *create_data->focused;
+#if defined(USE_ASH)
       saw_focus_key = true;
+#endif
     }
   }
 
-- 
2.7.4

